src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/model/sar/SARLicenceData.kt

```diff
@@ -0,0 +1,142 @@
+  val id: Long? = null,
+  var prisonNumber: String?,
+  val bookingId: Long,
+  val timestamp: LocalDateTime,
+  val version: Int,
+  val template: String,
+  val varyVersion: Int,
+  var deletedAt: LocalDateTime?,
+  var licenceInCvl: Boolean,
+  val licence: SARLicenceData?,
+)
+
+@JsonInclude(NON_NULL)
+data class SARLicenceData(
+  val eligibility: Eligibility?,
+  val bassReferral: Cas2Referral?,
+  val proposedAddress: ProposedAddress?,
+  val curfew: Curfew?,
+  val risk: Risk?,
+  val reporting: Reporting?,
+  val victim: Victim?,
+  val licenceConditions: SARConditions?,
+  val document: Document?,
+  val approval: Approval?,
+  val finalChecks: FinalChecks?,
+)
+
+@JsonInclude(NON_NULL)
+data class SARConditions(
+  val bespokeConditions: List<BespokeCondition>? = null,
+  val additionalConditions: List<SARAdditionalCondition>? = null,
+  val additionalConditionsJustification: String? = null,
+)
+
+@JsonInclude(NON_NULL)
+data class SARAdditionalCondition(
+  val text: String? = null,
+  val fields: Map<String, Any>? = null,
+)
+
+fun Licence.toSAR() = SARLicence(
+  id = this.id,
+  prisonNumber = this.prisonNumber,
+  bookingId = this.bookingId,
+  stage = this.stage,
+  version = this.version,
+  transitionDate = this.transitionDate,
+  varyVersion = this.varyVersion,
+  additionalConditionsVersion = this.additionalConditionsVersion,
+  standardConditionsVersion = this.standardConditionsVersion,
+  deletedAt = this.deletedAt,
+  licenceInCvl = this.licenceInCvl,
+  licence = this.licence?.toSAR(additionalConditionsVersion),
+)
+
+fun LicenceVersion.toSAR(): SARLicenceVersion = SARLicenceVersion(
+  id = this.id,
+  prisonNumber = this.prisonNumber,
+  bookingId = this.bookingId,
+  timestamp = this.timestamp,
+  version = this.version,
+  template = this.template,
+  varyVersion = this.varyVersion,
+  deletedAt = this.deletedAt,
+  licenceInCvl = this.licenceInCvl,
+  licence = this.licence?.toSAR(),
+)
+
+fun LicenceData.toSAR(conditionVersion: Int? = null) = SARLicenceData(
+  eligibility = this.eligibility,
+  bassReferral = this.bassReferral,
+  proposedAddress = this.proposedAddress,
+  curfew = this.curfew,
+  risk = this.risk,
+  reporting = this.reporting,
+  victim = this.victim,
+  licenceConditions = this.licenceConditions?.let {
+    SARConditions(
+      bespokeConditions = it.bespoke,
+      additionalConditions = it.additional?.entries?.map { (code, values) ->
+        toAdditionalCondition(
+          conditionVersion ?: attemptToGuessVersion(it.additional),
+          code,
+          values,
+        )
+      },
+      additionalConditionsJustification = it.conditionsSummary?.additionalConditionsJustification,
+    )
+  },
+  document = this.document,
+  approval = this.approval,
+  finalChecks = this.finalChecks,
+)
+
+private fun attemptToGuessVersion(additional: Map<String, Map<String, Any>>?): Int? = policyVersions.getPolicyVersion(additional?.map { condition -> condition.key } ?: emptyList())
```

Comment:
tries to guess what policy version it is by looking at condition ids on the licence and checking to see if any uniquely appear on one version of a policy. If so that's the badger.

It ignores conditions that appear on more than one policy version
It blows up if there are conditions from different policy versions on the same licence